### PR TITLE
feat: support leafCaseStyle option

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

for example, `app/model/admin/role.js -> app.model.admin.Role`

### CODE REVIEW

File: lib/loader/file_loader.js [hunk 1]
```diff
@@ -152,7 +153,7 @@ class FileLoader {
         const exports = getExports(fullpath, this.options, pathName);
 
         // ignore exports when it's null or false returned by filter function
-        if (exports == null || (filter && filter(exports) === false)) continue;
```
Comment (atian25) [inline lib/loader/file_loader.js]: 这里也顺便把 meta 传给 filter

File: lib/loader/file_loader.js [hunk 2]
```diff
@@ -221,31 +223,40 @@ function getExports(fullpath, { initializer, call, inject }, pathName) {
...
-      case 'camel':
-      default:
-    }
-    return first + property.substring(1);
+    return changeCase(property, options.caseStyle);
   });
+
+  if (options.leafCaseStyle) {
+    const lastIndex = properties.length - 1;
+    result[ lastIndex ] = changeCase(properties[lastIndex], options.leafCaseStyle);
```
Comment (popomore) [inline lib/loader/file_loader.js]: 那如果再多一层呢？

Comment (atian25) [reply lib/loader/file_loader.js]: `package.module.class` 的方式，前面几个一般都是小写的，就最后的 Class 需要大写

Comment (popomore) [reply lib/loader/file_loader.js]: 为啥要这样写

Comment (atian25) [reply lib/loader/file_loader.js]: 类似 model 这样的吧， alibaba.biz.User

Comment (atian25) [reply lib/loader/file_loader.js]: 这块可以先 hold，@XadillaX https://github.com/eggjs/egg/issues/1775 一起考虑

**Pull request closed without merging**